File: 2551. Put Marbles in Bags.py
class Solution:
    def putMarbles(self, A, K):
        if K == 1: return 0
        B = sorted([A[i] + A[i + 1] for i in range(len(A) - 1)])
        return sum(B[-K+1:]) - sum(B[:K-1])

File: 2551. Put Marbles in Bags.java
class Solution:
    def putMarbles(self, weights: List[int], k: int) -> int:
        pairCosts = [0] * (len(weights)-1)
        for i in range(len(weights) - 1):
            pairCosts[i] = weights[i] + weights[i + 1]
        pairCosts.sort()
        ans = 0;
        for i in range(k - 1):
            ans += pairCosts[len(pairCosts) - 1 - i] - pairCosts[i]

        return ans
